bundles/core/org.eclipse.smarthome.core.thing/src/main/java/org/eclipse/smarthome/core/thing/internal/ChannelItemProvider.java

```diff
@@ -141,13 +145,24 @@ protected synchronized void modified(Map<String, Object> properties) {
         }
 
         if (enabled) {
-            logger.debug("Enabling channel item provider.");
-            for (ProviderChangeListener<Item> listener : listeners) {
-                for (Item item : getAll()) {
-                    listener.added(this, item);
+            Executors.newSingleThreadExecutor().submit(new Runnable() {
+                @Override
+                public void run() {
+                    // we wait until no further new links or items are announced in order to avoid creation of
+                    // items which then must be removed again immediately.
+                    while (lastUpdate > new Date().getTime() - 2000L) {
```

Comment:
I assume you would like to use nanoTime here, too (and perhaps TimeUnit.SECONDS.toNanos(2))